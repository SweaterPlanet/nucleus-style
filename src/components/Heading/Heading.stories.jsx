import React from 'react'
import { withKnobs, number, select, text } from '@storybook/addon-knobs/react'
import Heading from '.'
import H1 from './H1'

export default {
  title: 'Heading',
  decorators: [withKnobs],
}

const textLabel = 'Text'
const textDefault = 'Heading 1'

const colorLabel = 'Color'
const colorOptions = {
  aqua: 'aqua',
  black: 'black',
  blue: 'blue',
  green: 'green',
  paper: 'paper',
  purple: 'purple',
  red: 'red',
  slate: 'slate',
  white: 'white',
}
const colorDefault = 'black'

const opacityLabel = 'Opacity'
const opacityDefault = '1'

export const asH1 = () => (
  <H1
    color={select(colorLabel, colorOptions, colorDefault)}
    opacity={number(opacityLabel, opacityDefault)}
  >
    {text(textLabel, textDefault)}
  </H1>
)

asH1.story = {
  name: 'as H1',
}
