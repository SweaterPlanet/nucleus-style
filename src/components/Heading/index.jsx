import React from 'react'
import { H1, H2, H3, H4 } from '/src'

const components = {
  h1: H1,
  h2: H2,
  h3: H3,
  h4: H4,
}

function Heading(props) {
  const HeadingByElement = components[props.element]
  return <HeadingByElement />
}

export default Heading
