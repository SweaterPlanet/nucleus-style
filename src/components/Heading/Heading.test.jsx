import React from 'react'
import { shallow } from 'enzyme'

import Heading from '.'

describe('Heading', () => {
  const headingElements = ['h1', 'h2', 'h3', 'h4']

  it('renders correctly', () => {
    const test = shallow(<Heading element="h1">Test</Heading>)
    expect(test).toMatchSnapshot()
  })

  headingElements.map(element => {
    it(`renders as ${element}`, () => {
      const test = shallow(<Heading element={element}>Test</Heading>)

      expect(test).toMatchSnapshot()
    })
  })

  it('respects the color prop', () => {
    const testProp = 'aqua'
    const test = shallow(<Heading element="h1" color={testProp} />)
    expect(test.prop('color')).toBe(testProp)
  })

  it('respects the opacity prop', () => {
    const testProp = 0.5
    const test = shallow(<Heading element="h1" opacity={testProp} />)
    expect(test.prop('opacity')).toBe(testProp)
  })
})
