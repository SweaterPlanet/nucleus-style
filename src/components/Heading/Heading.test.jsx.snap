// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders as h1 1`] = `ShallowWrapper {}`;

exports[`Heading renders as h2 1`] = `ShallowWrapper {}`;

exports[`Heading renders as h3 1`] = `ShallowWrapper {}`;

exports[`Heading renders as h4 1`] = `ShallowWrapper {}`;

exports[`Heading renders correctly 1`] = `ShallowWrapper {}`;
